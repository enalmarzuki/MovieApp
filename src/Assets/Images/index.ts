import IMGNotPhoto from './ic-not-image.png';

export {IMGNotPhoto};
