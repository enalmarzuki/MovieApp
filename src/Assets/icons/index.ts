import ICVideoCamera from './ic-video-camera.png';
import ICClock from './ic-clock.png';
import ICRating from './ic-rating.png';
import ICArrowLeft from './ic-arrow-left.png';

export {ICVideoCamera, ICClock, ICRating, ICArrowLeft};
