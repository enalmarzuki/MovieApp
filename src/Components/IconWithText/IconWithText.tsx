import {StyleSheet, Text, View, ImageSourcePropType, Image} from 'react-native';
import React from 'react';
import {fonts} from '../../Utils/fonts';
import {colors} from '../../Utils/colors';

interface IIconWithTextProps {
  icon: ImageSourcePropType;
  text: string;
}

const IconWithText: React.FC<IIconWithTextProps> = ({icon, text}) => {
  return (
    <View style={styles.iconWithTextWrapper}>
      <Image source={icon} testID="data-icon" />
      <Text style={styles.iconText} testID="data-text">
        {text}
      </Text>
    </View>
  );
};

export default IconWithText;

const styles = StyleSheet.create({
  iconWithTextWrapper: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  iconText: {
    fontFamily: fonts.primary.Regular,
    fontSize: 14,
    color: colors.placeholder,
  },
});
