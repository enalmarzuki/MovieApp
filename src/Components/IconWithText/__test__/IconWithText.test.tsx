import {ICArrowLeft} from '../../../Assets/icons';
import {render} from '@testing-library/react-native';
import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';
import IconWithText from '../IconWithText';
jest.useFakeTimers();

describe('should render ButtonBackWithIcon', () => {
  it('should render correctly', () => {
    const IconWithTextSnap = renderer
      .create(<IconWithText text="mock text" icon={ICArrowLeft} />)
      .toJSON();
    expect(IconWithTextSnap).toMatchSnapshot();
  });

  it('IconWithText must have text Synopsis', () => {
    const {getByTestId} = render(
      <IconWithText text="Synopsis" icon={ICArrowLeft} />,
    );

    const textContent = getByTestId('data-text').children[0];

    expect(textContent).toEqual('Synopsis');
    expect(getByTestId('data-icon')).toBeTruthy();
  });
});
