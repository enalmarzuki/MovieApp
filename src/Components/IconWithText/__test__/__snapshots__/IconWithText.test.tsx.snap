// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ButtonBackWithIcon should render correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }
>
  <Image
    source={
      {
        "testUri": "../../../src/Assets/icons/ic-arrow-left.png",
      }
    }
    testID="data-icon"
  />
  <Text
    style={
      {
        "color": "#BBBBBB",
        "fontFamily": "Lato-Regular",
        "fontSize": 14,
      }
    }
    testID="data-text"
  >
    mock text
  </Text>
</View>
`;
